code/modules/mob/living/carbon/human/human_stripping.dm

```diff
@@ -131,9 +131,25 @@ GLOBAL_LIST_INIT(strippable_human_items, create_strippable_list(list(
 	var/obj/item/clothing/under/uniform = sourcemob.w_uniform
 	if(!LAZYLEN(uniform.accessories))
 		return FALSE
-	var/obj/item/clothing/accessory/accessory = LAZYACCESS(uniform.accessories, 1)
+
+	var/obj/item/clothing/accessory/accessory
+	var/list/removables = list()
+	var/list/choice_to_accessory = list()
+
+	for(var/obj/item/clothing/accessory/acc in uniform.accessories)
+		if(!acc.removable)
+			continue
+		var/capitalized_name = capitalize_first_letters(acc.name)
+		removables[capitalized_name] = image(icon = acc.icon, icon_state = acc.icon_state)
+		choice_to_accessory[capitalized_name] = acc
+
 	if(LAZYLEN(uniform.accessories) > 1)
-		accessory = tgui_input_list(user, "Select an accessory to remove from [uniform]", "Remove accessory", uniform.accessories)
+		var/use_radials = usr.client.prefs?.no_radials_preference ? FALSE : TRUE
+		var/choice = use_radials ? show_radial_menu(user, source, removables, require_near = TRUE) : tgui_input_list(user, "Select an accessory to remove from [src]", "Remove accessory", removables)
+		accessory = choice_to_accessory[choice]
+	else
+		accessory = choice_to_accessory[removables[1]]
+
```

Comment:
Add a check here checking for if the target adjacent. Yeah radials have require_near but with tgui you can abuse this by calling the menu when adjacent and then choosing an accessory when not